### PR TITLE
[SYCLomatic] Implement a simple option to apply all SYCL extensions/experimental features during migration

### PR DESCRIPTION
**Pull request by yihwang-nv**
Add 3 options:
- `--use-experimental-features=all`
- `--use-dpcpp-extensions=all`
- `--no-dpcpp-extensions=all`

### CODE REVIEW

**Review by ziranzha — APPROVED** [review]

File: clang/include/clang/DPCT/DPCTOptions.inc [hunk 1]
```diff
@@ -378,6 +381,10 @@ DPCT_ENUM_OPTION(DPCT_OPT_TYPE(static llvm::cl::bits<DPCPPExtensionsDefaultDisab
                                      int(DPCPPExtensionsDefaultDisabled::ExtDD_IntelDeviceMath),
                                      "Use sycl::ext::intel::math functions from the libdevice library (provided by Intel(R) oneAPI\n"
                                      "DPC++/C++ Compiler) to migrate functions which have no mapping in the SYCL standard.",
+                                     false),
+             DPCT_OPT_ENUM("all",
+                                     int(DPCPPExtensionsDefaultDisabled::ExtDD_All),
+                                     "Enable all DPC++ extensions.",
```
Comment (zhimingwang36) [inline clang/include/clang/DPCT/DPCTOptions.inc]:
```suggestion
                                     "Enable all DPC++ extensions listed in this option.",
```

File: clang/include/clang/DPCT/DPCTOptions.inc [hunk 2]
```diff
@@ -423,6 +430,9 @@ DPCT_ENUM_OPTION(DPCT_OPT_TYPE(static llvm::cl::bits<ExperimentalFeatures>), Exp
         false),
              DPCT_OPT_ENUM("non-uniform-groups", int(ExperimentalFeatures::Exp_NonUniformGroups),
         "Experimental extension that allows use of non-uniform groups.\n",
+        false),
+            DPCT_OPT_ENUM("all", int(ExperimentalFeatures::Exp_All),
+        "Enable all experimental extensions.\n",
```
Comment (zhimingwang36) [inline clang/include/clang/DPCT/DPCTOptions.inc]:
```suggestion
        "Enable all experimental extensions listed in this option.\n",
```

File: clang/include/clang/DPCT/DPCTOptions.inc [hunk 3]
```diff
@@ -338,27 +338,30 @@ DPCT_ENUM_OPTION(DPCT_OPT_TYPE(static llvm::cl::bits<DPCPPExtensionsDefaultEnabl
...
+                                     "Disable the assert extension.", false),
              DPCT_OPT_ENUM("queue_empty",
                                      int(DPCPPExtensionsDefaultEnabled::ExtDE_QueueEmpty),
-                                     "Use the queue empty extension.",
-                                     false)
+                                     "Disable the queue empty extension.",
+                                     false),
+             DPCT_OPT_ENUM("all",
+                                     int(DPCPPExtensionsDefaultEnabled::ExtDE_All),
+                                     "Disable all extensions.", false)
```
Comment (zhimingwang36) [inline clang/include/clang/DPCT/DPCTOptions.inc]:
```suggestion
                                     "Disable all extensions listed in this option.", false)
```

Comment (yihwang-nv) [reply clang/include/clang/DPCT/DPCTOptions.inc]: done

File: clang/include/clang/DPCT/DPCTOptions.inc (hunk 2 above)
Comment (yihwang-nv) [reply clang/include/clang/DPCT/DPCTOptions.inc]: done

File: clang/include/clang/DPCT/DPCTOptions.inc (hunk 1 above)
Comment (yihwang-nv) [reply clang/include/clang/DPCT/DPCTOptions.inc]: done

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**